### PR TITLE
Add support for Gruntfile.babel.js

### LINKED ISSUE #1346: Support gruntfile.babel.js
See: https://github.com/gulpjs/gulp/issues/830

### PR DESCRIPTION
**Pull request by thejameskyle**
Resolves #1346

### CODE REVIEW

**Comment by michaelBenin** [comment]
Any news on this?

**Comment by thejameskyle** [comment]
ping

**Comment by vladikoff** [comment]
I added the "ready" tag to it so we shall look at this for the RC1 release

File: package.json [hunk 1]
```diff
@@ -35,6 +35,7 @@
   ],
   "dependencies": {
     "async": "~0.1.22",
+    "babel-core": "^5.8.23",
```
Comment (vladikoff) [inline package.json]: will need to lock this to ~

**Comment by silkentrance** [comment]
This limits ES/Babel sources to the extension '.babel.js'. Other peops are using for example '.es' or '.es6', which are the standard extensions. Add support for these extensions as well.

**Comment by shama** [comment]
I definitely want this as well but don't want to set a precedent for adding more language registers directly in Grunt at this time.

Going forward, we'll move towards [liftoff](https://github.com/js-cli/js-liftoff) which will enable support for additional languages and keeping dependencies more up to date. Look for an https://github.com/gruntjs/rfcs related to it soon (or open one if you would like to take on this endeavor sooner). Thanks!

**Comment by thejameskyle** [comment]
You've already set a precedent with coffeescript which is exactly what this PR was mimicking, but whatever

**Comment by shama** [comment]
@thejameskyle The existing coffeescript register is exactly why I don't want to add any more. With the language baked into Grunt, it becomes an API we've got to support. Users have been stuck with an old version of coffeescript because of that.

The language and version should be decided by the end user and not baked into Grunt. I'm a fan of Babel but if merging this as is would mean repeating that mistake and opening the door for more language registers to be baked in. I don't want that.

If you want to help, please sponsor a [rfc](https://github.com/gruntjs/rfcs) to implement support in a better way for Grunt and it's users.

**Comment by michaelBenin** [comment]
@shama you saw gulp has this feature right?

**Comment by timmywil** [comment]
@michaelBenin That was a painpoint for me when babel changed everything in babel 6. I ended up bypassing gulp's built-in version of babel by registering babel myself. It wasn't difficult, but it did require 2 files.

gulpfile.js

```js
require("babel-register")
require("./gulpfile.es")
```

gulpfile.es
```
// All the gulp stuff with whatever babel presets/plugins I want
```

I wonder if it'd be better to have some mechanism for requiring a module – whether it be coffee-script, babel, or really anything – before requiring the gulpfile.

**Comment by timmywil** [comment]
By the way, what I'm suggesting is akin to what mocha does already with the `--require` option in a mocha.opts file.

**Comment by dmethvin** [comment]
Yeah, I think that is where @shama was headed with the proposal, and the Babel 5/6 example is exactly what you want to avoid. Otherwise Grunt has a problem of quickly-changing dependencies and needs to do semver-major breaking releases on a regular basis. Ain't nobody got time for that.

**Comment by shama** [comment]
Funny enough, gulp uses Liftoff: https://github.com/gulpjs/gulp-cli/blob/master/package.json#L38 a library originally created by @tkellen for Grunt. So yes, I did see that gulp has this feature and is why I suggested it.

**Comment by cowboy** [comment]
FWIW, this is how I'm currently using Grunt with babel in a project:

https://github.com/bocoup/react-router-oauth/blob/master/Gruntfile.js
https://github.com/bocoup/react-router-oauth/blob/master/tools/gruntfile.js

**Comment by cowboy** [comment]
We had talked about this in the past, but if Grunt supported a `.gruntrc` file with options that got parsed before the `Gruntfile` was read, a `require` option could be specified like `{"require":["babel-register"]}` to tell Grunt to simply require every module in the array before doing anything else. Then the only thing hard-coded into Grunt would be the ability to load arbitrary modules.

**Comment by shama** [comment]
RFC for this started: https://github.com/gruntjs/rfcs/pull/2

**Comment by taoeffect** [comment]
Found it to be a bit more involved than @cowboy's Gruntfile to use `async`/`await`. Here's ours:

https://github.com/okTurtles/group-income-simple/blob/55cedb3e11b1d3873a234735b8e4c62f1a4defc6/Gruntfile.js

**Pull request closed without merging**